src/igd_async.rs

```diff
@@ -170,13 +170,10 @@ pub fn get_any_address(
 /// IPv4 because gateway always has IPv4 address as well.
 fn discover_local_addr_to_gateway(gateway_addr: Ipv4Addr) -> io::Result<Ipv4Addr> {
     let ifs = get_if_addrs::get_if_addrs()?;
-    local_addr_to_gateway(ifs, gateway_addr).map_or(
-        Err(io::Error::new(
-            io::ErrorKind::NotFound,
-            "No local addresses to gateway",
-        )),
-        |addr| Ok(addr),
-    )
+    let addr = local_addr_to_gateway(ifs, gateway_addr).ok_or_else(|| {
+        io::Error::new(io::ErrorKind::NotFound, "No local addresses to gateway")
+    })?;
+    Ok(addr)
```

Comment:
Redundant expression.
We could simply use `local_addr_to_gateway(...).ok_or_else(...)` as return value.